metricbeat/module/nats/varz/data.go

```diff
@@ -0,0 +1,150 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package varz
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+
+	"strconv"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	s "github.com/elastic/beats/libbeat/common/schema"
+	c "github.com/elastic/beats/libbeat/common/schema/mapstriface"
+)
+
+var (
+	httpReqStatsSchema = s.Schema{
+		"root_uri":   c.Int("/"),
+		"connz_uri":  c.Int("/connz"),
+		"routez_uri": c.Int("/routez"),
+		"subsz_uri":  c.Int("/subsz"),
+		"varz_uri":   c.Int("/varz"),
+	}
+	varzSchema = s.Schema{
+		"server_id":         c.Str("server_id"),
+		"now":               c.Str("now"),
+		"uptime":            c.Str("uptime"),
+		"mem":               c.Int("mem"),
+		"cores":             c.Int("cores"),
+		"cpu":               c.Int("cpu"),
+		"total_connections": c.Int("total_connections"),
+		"remotes":           c.Int("remotes"),
+		"in.msgs":           c.Int("in_msgs"),
```

Comment:
Can you convert this to 

```
"id": c.Dict("in", s.Schema{
  "messages": c.Int("in_msgs")
  "bytes": c.Int("in_bytes")
}
```

This will make sure it's already an object when leaving Metricbeat and not only converted in Elasticsearch.

The other change I made is `msgs` to `message` as we try to avoid abbrevations: https://www.elastic.co/guide/en/beats/devguide/current/event-conventions.html#event-conventions

The same applies to the `out` fields below. 

We are hoping to make this conversion easier and more straight forward in the future.

Reply:
Nice! What I don't get is the `"id"` key... did you go for `"in"` maybe, or I'm losing something?

Reply:
Type on my end, sorry.